Sources/Sentry/SentryFileManager.m

```diff
@@ -26,10 +28,14 @@
 
 @implementation SentryFileManager
 
-- (_Nullable instancetype)initWithDsn:(SentryDsn *)dsn didFailWithError:(NSError **)error
+- (_Nullable instancetype)initWithDsn:(SentryDsn *)dsn
+                  currentDateProvider:(id<SentryCurrentDateProvider>)currentDateProvider
```

Comment:
```suggestion
                  withCurrentDateProvider:(id<SentryCurrentDateProvider>)currentDateProvider
```

Reply:
M: I think the pattern used is either `withSomething` or `andSomething` If I'm not mistaken